app/components/layout/TabbedPage.js

```diff
@@ -103,7 +103,7 @@ function TabbedPage(props) {
   const [ dir, setDir ] = useState("l2r");
   useEffect( () => {
     setMatchedTab(getMatchedTab(location, props.children));
-  }, []);
+  }, [ props.children, location ]);
```

Comment:
same here

Reply:
Here is more case dependant, if the deps break the code and you have a really good case for keeping the dep list empty it's worth disabling the lint rule inline with `// eslint-disable-next-line react-hooks/exhaustive-deps`. Sometimes is misleading disabling it, tho. Take a look at the links in [this comment](https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528).

Reply:
I think we shouldn't disable the rule, I think we either should specify array with *all* deps, or with no array => will run one time like componentdidmount AFAIR.

Reply:
Here are some insights why we should be strict with the deps rule everywhere:
https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies
https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often

TLDR: we should break the useEffect to several ones instead of omitting deps, see second link above for more insights.

Reply:
Also, we shouldn't use deep properties like `props.children` in the deps array, instead we should always define another const outside the useEffect and use it instead inside the `useEffect` and in the deps array!

Reply:
Yeah, the useEffect hook should be used for a specific task and you can have multiple useEffect hooks. Even better if you have custom hooks with descriptive names for each of these tasks if they are related or just too long.